package.json

```diff
@@ -80,23 +84,26 @@
     "style-loader": "3.3.3",
     "styled-components": "^6.1.0",
     "swc-loader": "^0.2.3",
-    "ts-node": "^10.9.1",
+    "terser-webpack-plugin": "^5.3.10",
+    "ts-node": "^10.9.2",
     "tsconfig-paths": "^4.2.0",
     "tslib": "2.5.3",
-    "typescript": "5.3.3",
-    "webpack": "^5.86.0",
+    "typescript": "5.5.4",
+    "webpack": "^5.94.0",
     "webpack-cli": "^5.1.4",
     "webpack-livereload-plugin": "^3.0.2",
-    "webpack-merge": "^5.8.0"
+    "webpack-merge": "^5.8.0",
+    "webpack-subresource-integrity": "^5.1.0",
+    "webpack-virtual-modules": "^0.6.2"
   },
   "dependencies": {
-    "@emotion/css": "^11.1.3",
-    "@grafana/data": "11.0.0",
+    "@emotion/css": "11.10.6",
+    "@grafana/data": "^11.3.0",
     "@grafana/faro-web-sdk": "1.3.6",
-    "@grafana/runtime": "11.0.0",
+    "@grafana/runtime": "^11.0.0",
     "@grafana/scenes": "5.1.0",
     "@grafana/schema": "11.0.0",
-    "@grafana/ui": "11.0.0",
+    "@grafana/ui": "^11.0.0",
```

Comment:
Sorry, I should have been clearer. I am all for updating the packages -- little and often is great and if we can continually update the Grafana packages especially so they are as close to the runtime as possible that's great.

My concern is in utilising the `^` caret syntax for the dependencies. I'm not a fan of it because `package.json` becomes opaque and you have to delve into `yarn.lock` to know what you are really using. It also assumes that every package you utilise accurately follows semver and I have trust issues after this being a source of bugs in other projects I have worked on previously 😅

Reply:
I can see why pinning `@grafana/*` versions makes sense (even though it doesn't mean that the code will run with that version in the wild), but pinning "other" packages for no other reason than "what if..." seems like a false sense of security. Many of our dependencies have their own dependencies that will install `minor` and `patch`, meaning that we have no actual control over potential side-effects by installing a minor/patch.
Also, when two (including our package) or more packages rely on the same major version, they may share the same package instead of installing their specific version.

Note: is `bundleGrafanaUI: false` something we want to use? I think it ties into this.

TL;DR: I could live with (but, wouldn't recommend) pinned versions with `~` (so that patches are installed).

Reply:
Alright, let's agree to pin the runtime `@grafana` packages and update the plugin.json grafanaDependency to match.

We haven't encountered any problems with the caret syntax and you are right, it does help manage the dependency tree better.

> Note: is bundleGrafanaUI: false something we want to use? I think it ties into this.

No, I don't think we do. It just creates a different set of problems and goes against the grain of what everyone else is doing. Because we have SLOs with CVEs now we should naturally get a pretty good rhythm of keeping our dependencies updated.